helix-view/src/document.rs

```diff
@@ -928,9 +928,14 @@ impl Document {
                 }
             }
 
-            // Protect against overwriting changes made externally
             if !force {
                 if let Ok(metadata) = fs::metadata(&path).await {
+                    // Protect against overwriting entire directories
+                    // if `path` doesn't end with a directory separator.
+                    if metadata.is_dir() {
```

Comment:
Thus should probably be `!is_file()` instead to cover esgecases like pipes

Reply:
Makes sense... and how about symlinks? Should Helix abort writing when the destination is a symlink? I would expect it to work if I specify a filename that is a symlink to a regular file... but checking for that would complicate thing quite a bit...

Reply:
If its a synlink metadata will return information about the destination of the symlink not the symlink itself.

And similarly helix will write to the destination of the symlink not the symli k itself. So they are treated transparently (this is the standard across editors)

Reply:
You are right, I missed this part from [the docs](https://doc.rust-lang.org/std/fs/fn.metadata.html):

> This function will traverse symbolic links to query information about the destination file.

I changed the condition to `!metadata.is_file()`,